File: alexandria/ui/src/io/intino/alexandria/ui/displays/components/BlockParallax.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.resources.Asset;

import java.net.URL;

public class BlockParallax<B extends Box> extends AbstractBlockParallax<B> {
    private URL background;

    public BlockParallax(B box) {
        super(box);
    }

    @Override
    public void init() {
        super.init();
        refresh();
    }

    public String background() {
        if (background == null) return null;
        return Asset.toResource(baseAssetUrl(), background).toUrl().toString();
    }

    public BlockParallax background(URL background) {
        this.background = background;
        return this;
    }

    public void refresh() {
        String background = background();
        if (background == null) return;
        notifier.refresh(background);
    }

}

File: web-ui/src/displays/components/BlockParallax.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractBlockParallax from "../../../gen/displays/components/AbstractBlockParallax";
import BlockParallaxNotifier from "../../../gen/displays/notifiers/BlockParallaxNotifier";
import BlockParallaxRequester from "../../../gen/displays/requesters/BlockParallaxRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';

const styles = theme => ({});

class BlockParallax extends AbstractBlockParallax {

	constructor(props) {
		super(props);
		this.notifier = new BlockParallaxNotifier(this);
		this.requester = new BlockParallaxRequester(this);
	};

	refresh = (value) => {
	};
}

export default withStyles(styles, { withTheme: true })(BlockParallax);
DisplayFactory.register("BlockParallax", withStyles(styles, { withTheme: true })(BlockParallax));